### PR TITLE
Add delete feature

### PR DESCRIPTION
**Pull request by sasfmlzr**
This feature removes some items from the task list.

For ToDoRepository was added coroutine scope to the constructor. If we use GlobalScope, we won't be able to catch any exceptions.
Also, FRACTION_OF_TIME for coroutine tests was increased because I couldn't get the stability of the tests. 

**Type of change:**
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**
- FakeToDoRepositoryTest (e.g I performed removing elements from FakeToDoRepository)

**Checklist:**
- [x] This PR follows [guidelines](https://github.com/DistilleryTech/android-arch-blueprints/wiki/Pull-Request-Process)
- [x] I have performed a self-review of my own code
- [x] I documented all new code and updated the external docs if necessary
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by 0neel** [comment]
Note that all the implementation details should go to the first description part. Edited it for this time.

File: domain/src/main/java/com/distillery/android/domain/FakeToDoRepository.kt [hunk 1]
```diff
@@ -122,4 +122,11 @@ class FakeToDoRepository : ToDoRepository {
         }
         todosChannel.close(IllegalArgumentException("You died"))
     }
+
+    override suspend fun deleteToDo(uniqueId: Long) {
+        delay(DELAY_FOR_VALUE_ADDITION)
```
Comment (0neel) [inline domain/src/main/java/com/distillery/android/domain/FakeToDoRepository.kt:127]: The constant name doesn't correspond to its purpose anymore. It's better to either add another constant for the deletion or make the name more generic, like this:
```suggestion
        delay(DELAY_FOR_TODO_OPERATION)
```

File: domain/src/test/java/com/distillery/android/domain/FakeToDoRepositoryTest.kt [hunk 2]
```diff
@@ -73,6 +75,34 @@ class FakeToDoRepositoryTest {
         subscription.cancel()
     }
 
+    @Test
+    fun `repository delete values after 1_5 seconds`() = runBlocking {
+        // given we subscribed to repository changes
```
Comment (0neel) [inline domain/src/test/java/com/distillery/android/domain/FakeToDoRepositoryTest.kt:80]: Minor: I think such verbose comments aren't necessary
```suggestion
        // GIVEN
```

File: domain/src/test/java/com/distillery/android/domain/FakeToDoRepositoryTest.kt [hunk 3]
```diff
@@ -65,7 +67,33 @@ class FakeToDoRepositoryTest {
...
+                listSize = it.size
+            }
+        }
+        check(listSize == 0)
+
+        // when enough time has passed 2 elements were added to list
+        delay(DELAY_OF_VALUES_GENERATOR * EXPECTED_AMOUNT_OF_ELEMENTS + FRACTION_OF_TIME)
+
+        // when we delete element
+        launch {
```
Comment (distillery-alexander-ageychenko) [inline domain/src/test/java/com/distillery/android/domain/FakeToDoRepositoryTest.kt:92]: is `launch` required here? AS I see we are already in a coroutine scope created by `runBlocking `

File: domain/src/test/java/com/distillery/android/domain/FakeToDoRepositoryTest.kt [hunk 4]
```diff
@@ -24,7 +25,8 @@ class FakeToDoRepositoryTest {
 
     @Before
     fun setup() {
-        repository = FakeToDoRepository()
+        val scope = GlobalScope
+        repository = FakeToDoRepository(scope)
```
Comment (distillery-alexander-ageychenko) [inline domain/src/test/java/com/distillery/android/domain/FakeToDoRepositoryTest.kt:29]: Minor: can be inlined
```suggestion
        val scope = GlobalScope
        repository = FakeToDoRepository(scope = GlobalScope)
```

**Pull request merged**